js/copy-service/Evaluator/Evaluator.ts

```diff
@@ -0,0 +1,109 @@
+import _ from 'lodash';
+
+import CopyService, { AST } from '../CopyService';
+import ErrorHandler from '../ErrorHandler/ErrorHandler';
+import IntlCopyService from '../IntlCopyService';
+import Substitutions from '../Substitutions/Substitutions';
+
+/**
+ * Provides an interface to recursively generate copy evaluated with substitutions.
+ *
+ * @interface
+ */
+class Evaluator {
+  /**
+   * The copy service used for evaluation.
+   *
+   * @type {CopyService|IntlCopyService}
```

Comment:
We should be able to axe these `@type` tags now. If JSDoc linter is throw, let's disable that rule. I disabled the rule for `@param` and `@returns`, but I may have missed `@type` and `@typedef`.